extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/GZipTest.java

```diff
@@ -0,0 +1,105 @@
+package io.quarkus.resteasy.reactive;
+
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.function.Supplier;
+
+import javax.imageio.ImageIO;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+
+public class GZipTest {
+
+    private static final String APP_PROPS = "" +
+            "quarkus.http.enable-compression=true\n";
+
+    static String longString;
+    static {
+
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < 1000; ++i) {
+            sb.append("Hello RESTEasy Reactive;");
+        }
+        longString = sb.toString();
+    }
+
+    @RegisterExtension
+    static final QuarkusUnitTest test = new QuarkusUnitTest()
+            .setArchiveProducer(new Supplier<JavaArchive>() {
+                @Override
+                public JavaArchive get() {
+                    return ShrinkWrap.create(JavaArchive.class)
+                            .addAsResource(new StringAsset(APP_PROPS), "application.properties")
+                            .addClasses(TestCompression.class);
+                }
+            });
+
+    @Test
+    public void testServerCompression() throws Exception {
+
+        RestAssured.given().get("/test/compression").then().statusCode(200)
+                .header("content-encoding", "gzip")
+                .header("content-length", Matchers.not(Matchers.equalTo(Integer.toString(longString.length()))))
+                .body(Matchers.equalTo(longString));
+
+        RestAssured.given().get("/test/nocompression").then().statusCode(200)
+                .header("content-encoding", "identity")
+                .header("content-length", Matchers.equalTo((long) (createImage().getData().getDataBuffer().getSize() * 4L)))
```

Comment:
```
(long) (createImage().getData().getDataBuffer().getSize() * 4L)
```

Finding the `content-length` for `BufferedImage` is something I am not much sure about. 
I took reference from here - 
https://stackoverflow.com/questions/632229/how-to-calculate-java-bufferedimage-filesize?noredirect=1&lq=1

Reply:
Humm I thought it's better to generate an image only when this test is run. 

You are right, it will make code simpler. Instead of creating an image through code it is better to use a static image 💯

Reply:
Done. 

Original size - 2914 bytes
Compressed size - 2741 bytes
![image](https://user-images.githubusercontent.com/46818757/117913535-23164a00-b2ff-11eb-8920-1f252b0afd26.png)